packages/amount-selectors/src/components/AmountSelectors.js

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Button, { buttonTypes } from "@crave/farmblocks-button";
+import InputText from "@crave/farmblocks-input-text";
+import { fontSizes } from "@crave/farmblocks-theme";
+import values from "object.values";
+
+import selectorSizes from "../constants/selectorSizes";
+import Wrapper from "../styledComponents/AmountSelector";
+
+class AmountSelectors extends React.Component {
+  state = {
+    value: this.props.value,
+    displayValue: this.props.value,
+    validationMessages: this.props.validationMessages
+  };
+
+  onChange = (event, cb) => {
+    const value = typeof event === "number" ? event : event.target.value;
+    const parsedValue = parseFloat(value) || 0;
+
+    if (parsedValue >= this.props.min && parsedValue <= this.props.max) {
+      this.setState(
+        {
+          value: parsedValue,
+          validationMessages: []
+        },
+        cb
+      );
+
+      return this.props.onChange(parsedValue);
+    }
+
+    this.setState({
+      validationMessages: ["Invalid value"]
+    });
+  };
+
+  updateDisplayValue = () => {
+    this.setState({
+      displayValue: this.state.value.toFixed(2)
+    });
+  };
+
+  decrement = () => {
+    const value = this.state.value - this.props.step;
+
+    this.onChange(Math.max(value, this.props.min), this.updateDisplayValue);
+  };
+
+  increment = () => {
+    const value = this.state.value + this.props.step;
+
+    this.onChange(Math.min(value, this.props.max), this.updateDisplayValue);
+  };
+
+  render() {
+    return (
+      <Wrapper size={this.props.size}>
+        <Button
+          className="decreaseButton"
+          type={buttonTypes.SECONDARY}
+          size={this.props.size}
```

Comment:
🤔 The `Button` component receives a `selectorSizes` value? Don't buttons have their own constants `buttonSizes`?
They may be identical now but that could not be true in the future.

How about we create a helper function:
```javascript
const mapSelectorSizeToButtonSize = size => ({
  [selectorSizes.SMALL]: buttonSizes.SMALL,
  [selectorSizes.MEDIUM]: buttonSizes.MEDIUM
})[size];
```